Declaring Java interface deprecation (#2365)
oneapi-src/oneDAL: docs/source/daal-interfaces.rst

You can find examples on Github*:

- `C++ <https://github.com/oneapi-src/oneDAL/tree/master/examples/daal/cpp/source>`_ (CPU)

> - `Java* <https://github.com/oneapi-src/oneDAL/tree/master/examples/daal/java/com/intel/daal/examples>`_ (nDeprecation Notice: Java interfaces in the oneDAL library have been deprecated and may no longer be supported in future releases.)

Reviewer: ```suggestion
- `Java* <https://github.com/oneapi-src/oneDAL/tree/master/examples/daal/java/com/intel/daal/examples>`_ (Deprecation Notice: The Java interfaces are deprecated in the oneDAL library and may no longer be supported in future releases.)
```
